leshan-core/src/test/java/org/eclipse/leshan/core/request/execute/ArgumentValidationTest.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Orange.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Orange - Add better support for Arguments of Execute Operation.
+ *******************************************************************************/
+package org.eclipse.leshan.core.request.execute;
+
+import static org.junit.Assert.assertThrows;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.function.ThrowingRunnable;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class ArgumentValidationTest {
+
+    @Parameterized.Parameters(name = "{index} : {0}")
+    public static Collection<?> linkValueListProvider() {
+        return Arrays.asList( //
+                new TestParams(0, "hello", null), //
+                new TestParams(9, "hello", null), //
+                new TestParams(-1, "hello", InvalidArgumentException.class), //
+                new TestParams(10, "hello", InvalidArgumentException.class), //
+                new TestParams(1, null, null), //
+                new TestParams(1, "", null), //
+                new TestParams(1,
+                        "!" + new String(new byte[] { 0x23, 0x26, 0x28, 0x5B, 0x5D, 0x7E }, StandardCharsets.UTF_8),
+                        null), //
+                new TestParams(1, new String(new byte[] { 0x22 }, StandardCharsets.UTF_8),
+                        InvalidArgumentException.class), // " character
+                new TestParams(1, new String(new byte[] { 0x27 }, StandardCharsets.UTF_8),
+                        InvalidArgumentException.class), // ' character
+                new TestParams(1, new String(new byte[] { 0x5C }, StandardCharsets.UTF_8),
+                        InvalidArgumentException.class), // \ character
+                new TestParams(1, new String(new byte[] { 0x7F }, StandardCharsets.UTF_8),
+                        InvalidArgumentException.class), // DEL character
+                new TestParams(1, "`aAzZ190-=~!@#$%^&*()_+[]{}|;:<>/?,.", null) // more visualized character rules above
+        );
+    }
```

Comment:
It's OK to me if we keep it like this but I'm not sure if you create TestParams class on purpose or because you don't know you can pass several Parameterized.parameters ? 

So just in case : https://www.tutorialspoint.com/junit/junit_parameterized_test.htm

Reply:
I know, but I wanted to use toString() to pretty print test case name. But now it looks like quite complicated so I make it in standard way.

Reply:
You can also change the test name with : https://github.com/junit-team/junit4/wiki/Parameterized-tests#identify-individual-test-cases, but this will not be as flexible as a toString() method.